### ISSUE TITLE
Wonky background/linear-gradient minification

### ISSUE THREAD

**Issue by ngyikp** [jakubpawlowicz/clean-css#821]
Version 3.4.20

Sorry for the issue title/description, not sure how to explain this best. I reduced my test case to the most minimal I could:

``` css
.message.error {
  background-color: #000;
}
.message,
.message.error {
  background-color: #fff;
  background-image: url('x'), -webkit-linear-gradient(top, #aaa, #aaa);
  background-image: url('x'), -moz-linear-gradient(top, #aaa, #aaa);
  background-image: url('x'), linear-gradient(to bottom, #aaa, #aaa);
}
.message.error {
  background-image: url('x'), -webkit-linear-gradient(top, #bbb, #bbb);
  background-image: url('x'), -moz-linear-gradient(top, #bbb, #bbb);
  background-image: url('x'), linear-gradient(to bottom, #bbb, #bbb);
}
.message,
.message.error {
  overflow: visible;
}
```

Restructuring on/off makes no difference.

Output:

``` css
.message,.message.error{background-color:#fff;background-image:url(x),-webkit-linear-gradient(top,#aaa,#aaa);background-image:url(x),-moz-linear-gradient(top,#aaa,#aaa);background-image:url(x),linear-gradient(to bottom,#aaa,#aaa);overflow:visible}
.message.error{background-image:url(x),linear-gradient(to bottom,#bbb,#bbb)}
```

`.message.error` is missing the `-webkit` and `-moz` linear gradients.

**Comment by jakubpawlowicz** [comment]
You are absolutely right.

**Comment by jakubpawlowicz** [comment]
It's fixed in 3.4.21.
